dgraph/cmd/zero/zero.go

```diff
@@ -441,6 +441,12 @@ func (s *Server) Connect(ctx context.Context,
 	if err != nil {
 		return nil, err
 	}
+
+	if m.Learner && !ms.License.GetEnabled() {
+		// Update the "ShouldCrash" function in x/x.go if you change the error message here.
+		return nil, errors.New("Missing or expired Enterpise License. Cannot add Learner Node.")
```

Comment:
QQ, I am not sure if this will work. ShouldCrash check for `ENTERPRISE_ONLY_LEARNER `. We are not returning this in the error. I believe error message is incomplete. Am I missing anything?

Reply:
Thanks! Fixed it.